.github/workflows/test.yml

```diff
@@ -0,0 +1,18 @@
+name: CI
+
+on: pull_request
+
+concurrency:
+  group: ${{ github.ref_name }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  rubocop:
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+
+  test:
+    name: Tests
+    uses: theforeman/actions/.github/workflows/test-gem.yml@v0
```

Comment:
We have a dedicated Smart Proxy test workflow. Would you mind updating to that? Biggest difference is that we can enforce certain Ruby version support and fail checks if that isn't there.